app/uk/gov/hmrc/pillar2stubs/controllers/LiabilityController.scala

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.pillar2stubs.controllers
+
+import play.api.libs.json.{JsError, JsSuccess, Json}
+import play.api.mvc.{Action, ControllerComponents}
+import uk.gov.hmrc.pillar2stubs.controllers.actions.AuthActionFilter
+import uk.gov.hmrc.pillar2stubs.models.UKTRSubmissionRequest
+import uk.gov.hmrc.pillar2stubs.utils.ResourceHelper.resourceAsString
+import uk.gov.hmrc.play.bootstrap.backend.controller.BackendController
+import play.api.Logging
+
+import java.time.LocalDate
+import javax.inject.{Inject, Singleton}
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+@Singleton
+class LiabilityController @Inject() (
+  cc:          ControllerComponents,
+  authFilter:  AuthActionFilter
+)(implicit ec: ExecutionContext)
+    extends BackendController(cc)
+    with Logging {
+
+  def createLiability(plrReference: String): Action[String] = (Action(parse.tolerantText) andThen authFilter).async { implicit request =>
+    if (plrReference != "XTC01234123412") {
+      logger.warn(s"Invalid PLR Reference provided: $plrReference")
+      Future.successful(NotFound(Json.obj("error" -> "No liabilities found for the given reference")))
+    } else {
+
+      Try(Json.parse(request.body)) match {
+        case Success(json) =>
+          json.validate[UKTRSubmissionRequest] match {
+            case JsSuccess(_, _) =>
+              val successResponse = resourceAsString("/resources/liabilities/LiabilitySuccessResponse.json")
```

Comment:
a change in approach to the existing stubs; would it be worthwhile to parse this into a JSON object and manipulate it (rather than string replace)?

Reply:
actually, I've just realised this is the existing stub. Ignore this point; it's better to be consistent with other code